src/components/table/table.stories.js

```diff
@@ -26,12 +26,12 @@ const handleChange = (e, tableOptions) => {
   action('change')(e, tableOptions);
 };
 
-const buildRows = (pageSizeFromKnobs) => {
+const buildRows = (pageSizeFromKnobs, totalRecords) => {
   const pageSize = pageSizeFromKnobs;
   const currentPage = store.get('currentPage');
 
-  const endIndex = pageSize * currentPage;
-  const startIndex = endIndex - pageSize;
+  const endIndex = pageSize * currentPage <= totalRecords ? pageSize * currentPage : totalRecords;
+  const startIndex = totalRecords === endIndex ? totalRecords - (endIndex % pageSize) : endIndex - pageSize;
```

Comment:
Maybe this is a bit clearer?
```
let endIndex = pageSize * currentPage;
if (endIndex > totalRecords) endIndex = totalRecords;

const currentPageSize = (endIndex === totalRecords) ? (endIndex % pageSize) : pageSize;
const startIndex = endIndex - currentPageSize
```

Reply:
Gone for a halfway between your way and mine